src/Catalog/CatalogIndexEntry.cs

```diff
@@ -20,6 +21,17 @@ public class CatalogIndexEntry : IComparable<CatalogIndexEntry>
         private readonly string _id;
         private readonly NuGetVersion _version;
 
+        public CatalogIndexEntry(JToken item)
+            : this(
+                new Uri(item["@id"].ToString()),
+                item["@type"].ToString(),
+                item["commitId"].ToString(),
+                DateTime.Parse(item["commitTimeStamp"].ToString()),
```

Comment:
DateTime.Parse can throw (https://msdn.microsoft.com/en-us/library/1k1skd40(v=vs.110).aspx). Does this block behave correctly if it does?